File: src/kyu_4/next_smaller_number_with_the_same_digits/csharp/next_smaller_number_with_the_same_digits_test.cs
using NUnit.Framework;
[TestFixture]
public class Tests
{
  [TestCase(21, ExpectedResult = 12)]
  [TestCase(907, ExpectedResult = 790)]
  [TestCase(531, ExpectedResult = 513)]
  [TestCase(1027, ExpectedResult = -1)]
  [TestCase(441, ExpectedResult = 414)]
  [TestCase(123456798, ExpectedResult = 123456789)]
  public long FixedTests(long n)
  {
    return Kata.NextSmaller(n);
  }
}

File: src/kyu_4/next_smaller_number_with_the_same_digits/python/next_smaller_number_with_the_same_digits_test.py
import codewars_test as test
from solution import next_smaller
@test.describe('Fixed Tests')
def fixed_tests():
    @test.it('Simple Cases')
    def example_cases():
        test.assert_equals(next_smaller(907), 790)
        test.assert_equals(next_smaller(531), 513)
        test.assert_equals(next_smaller(135), -1)
        test.assert_equals(next_smaller(2071), 2017)
        test.assert_equals(next_smaller(414), 144)
        test.assert_equals(next_smaller(123456798), 123456789)
        test.assert_equals(next_smaller(123456789), -1)
        test.assert_equals(next_smaller(1234567908), 1234567890)